payment-processor/src/main/kotlin/org/ostelco/prime/paymentprocessor/StripeUtils.kt

```diff
@@ -23,29 +24,33 @@ object StripeUtils {
         return try {
             Either.right(action())
         } catch (e: CardException) {
-            // If something is decline with a card purchase, CardException will be caught
-            logger.warn("Payment error : $errorDescription , Stripe Error Code: ${e.code}")
-            Either.left(ForbiddenError(errorDescription, e.message))
+            /* Something is wrong with the payment source/card.
+               402 indicates that the card is out of funds or expired etc. */
+            logger.warn("Payment error: ${errorDescription}, Stripe Error Code: ${e.code}")
+            Either.left(if (e.statusCode == 402)
+                PaymentFailedError(errorDescription, e.message)
+            else
+                ForbiddenError(errorDescription, e.message))
```

Comment:
What is forbidden error? An error because something is forbidden, or an error that should never happen?   If it's the latter then the method should be given a better name,, otherwise some comment should indicate why the invocation was forbidden.